### ISSUE TITLE
IAT seems to slow down this web page a lot (~30 seconds)

### ISSUE THREAD

**Issue by dstndstn** [docwhat/itsalltext#50]
I'm a huge fan of It's All Text.  However, it seems to cause a huge slowdown with a web site I use all the time: the "Query" page of the CasJobs database for the Sloan Digital Sky Survey:
  http://skyservice.pha.jhu.edu/casjobs/SubmitJob.aspx
(unfortunately, this requires a (free) registration to login).

With IAT disabled, the page loads quickly.  With IAT enabled, it seems to begin loading, then I get the dreaded Mac Happy Beachball of Death spinning for 20-30 seconds, and then the page finally loads.

I don't know what information I can provide to help tracking this down.  (I also understand that this is a very obscure web site so fixing it is not going to be a priority!)

Sometimes I get the "Unresponsive script" dialog box, complaining about:
Script: chrome://itsalltext/content/cacheobj.js:206
or
Script: chrome://itsalltext/content/cacheobj.js:843
or other places.

Firefox 23.0
MacOS X 10.8.3
IAT 1.8.1

Here is a screenshot when I use the "profile" tool of the Firefox "Web Developer" tools.

![1](https://f.cloud.github.com/assets/1065981/948306/0b13849c-0365-11e3-855d-5649583ea816.png)

Thanks for any help you can provide.  I am also reporting this to the CasJobs team and will let you know if they have anything to say.
--dustin

**Comment by noah** [comment]
I have experienced the same issue ... the ability to selectively disable IAT on certain pages might help.

**Comment by docwhat** [comment]
@noah Cool!

So you can see checks for `isHTML()` and `isXUL()` in [`monitor.js`](https://github.com/docwhat/itsalltext/blob/master/src/chrome/content/monitor.js#L70).

You probably want to do something there like the `!doc` check at the top:

``` .js
if (isBlacklisted()) {
  return [];
}
```

**Comment by noah** [comment]
I think I've got something working, feel free to give it a try:

https://github.com/noah/itsalltext/commit/0e2d402366e1af073375949080ecca79f351b84c

Somewhat difficult to debug FF addons so I'm not 100% sure.

**Comment by docwhat** [comment]
Did you look in the `DEBUGGING.md` file?

**Comment by noah** [comment]
Nope, but I managed to arrive at equivalent steps independently.  Would have been helpful, though and bookmarked for future reference.
